pkg/container/device.go

```diff
@@ -108,8 +112,8 @@ func (c *container) specAddContainerConfigDevices() error {
 				Type:  string(dev.Type),
 				Major: dev.Major,
 				Minor: dev.Minor,
-				UID:   &dev.Uid,
-				GID:   &dev.Gid,
+				UID:   getDeviceUID(c.Config(), dev.Uid, disableDeviceOwnershipFromSecurityContext),
+				GID:   getDeviceGID(c.Config(), dev.Gid, disableDeviceOwnershipFromSecurityContext),
```

Comment:
nit: we missed a case below (where the mount is a directory of devices). we should probably be consistent
also, in doing so, we should save the device{G,U}Id to not redo the work